src/librustc_trait_selection/traits/fulfill.rs

```diff
@@ -515,6 +515,16 @@ impl<'a, 'b, 'tcx> ObligationProcessor for FulfillProcessor<'a, 'b, 'tcx> {
                     Err(err) => ProcessResult::Error(CodeSelectionError(ConstEvalFailure(err))),
                 }
             }
+
+            ty::Predicate::WellFormedConst(constant) => {
+                ProcessResult::Changed(mk_pending(wf::const_obligations(
+                    self.selcx.infcx(),
+                    obligation.param_env,
+                    obligation.cause.body_id,
+                    constant,
+                    obligation.cause.span,
+                )))
```

Comment:
Just like `wf::obligations`, you should have `Option`, where `None` indicates no progress (i.e. further inference is required.

Maybe it would be better to have a `struct NoProgress;` and `compute` return `Result<(), NoProgress>` instead of `bool`.

Reply:
used `Result</* */, NoProgress>` everywhere.